### ISSUE TITLE
Shortcut to cycle through open tabs

### ISSUE THREAD

**Issue by eakwarren** [musescore/MuseScore#23702]
### Your idea

Alt+Tab (Win) and Command+Tab cycles through apps in the OS. Shortcuts with the ~ key instead of Tab cycle between windows in the same app. Since MSS doesn't support multiple windows in an app instance, a shortcut to Alt+~ or Command+~ to cycle through open tabs (like different parts), would help navigating scores with a large number of parts. This may also be implemented as a previous tab/next tab shortcut pair in Preferences>Shortcuts..

### Problem to be solved

Currently in MSS v4.3.2, F6 cycles through some tabs (Palettes, Instruments, the Score, etc.) When the Score is highlighted, the left/right arrow keys next select the X button and then the next open tab, etc. This is cumbersome and unintuitive. In scores with many parts, mousing to the tab can be much slower than a prev/next keyboard shortcut.

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by cbjeukendrup** [comment]
I think we should not mess with the Cmd+` (or Cmd+~) shortcut at the moment, because the separate-instance-for-each-window situation won't last much longer. So then that shortcut gets back its meaning as intended by the OS.

For cycling through tabs, I would propose Ctrl+Tab and Ctrl+Shift+Tab. Just like in Safari. Cmd+Tab (Alt+Tab on Windows) switches between apps, Ctrl+Tab cycles through apps. (This is one of the exceptional cases where both Mac and non-Mac should use the Ctrl key, and not Cmd.)

**Comment by MarcSabatella** [comment]
For the record, Ctrl+Tab is how MU3 did it as well.  Also Chrome.  So, seems the no-brainer choice to me.

**Comment by eakwarren** [comment]
Ctrl+Tab and Ctrl+Shift+Tab would be great! (And I learned something new about Safari. Thanks! ;-)

**Comment by Eism** [comment]
I agree with Ctrl+Tab. As an addition, we can also implement Ctrl+number - switching to the tab by index. The same approach is used for Chrome. An interesting point is that Ctrl+9 opens the last tab.

Both shortcuts are used in the current shortcuts system: Ctrl+Tab in navigation - we can replace it, Ctrl+number in text navigation - this shortcut only works if text editing has started, so I don’t see any problems with additional behavior.

**Comment by MarcSabatella** [comment]
What is Ctrl+Tab meant to do currently?  Seems to be the same as Tab alone on my system (Linux).

**Comment by shoogle** [comment]
I support using Ctrl+Tab for switching tabs, but I think this should only happen while the score has focus. When focus is elsewhere in the UI (e.g. in the toolbars, palettes, Mixer, etc.), Ctrl+Tab should continue to act as it does now.

> What is Ctrl+Tab meant to do currently? Seems to be the same as Tab alone on my system (Linux).

It moves focus to the next navigation panel (i.e. the next control group), and you're right, that's also what Tab does without the Ctrl modifier.

We've had feedback from accessibility users and agencies that it's unintuitive for the Tab key to navigate only to control groups rather than to each individual control. Blind users rely on the Tab key to get around, so if it skips some controls then blind users won't know that those controls exist.

So I think we should make the Tab key visit every control like in a traditional app, and use Ctrl+Tab as the "go-faster key" to navigate directly between control groups. When the score has focus, Ctrl+Tab can navigate between tabs.

>  We can also implement Ctrl+number - switching to the tab by index

On Windows and Linux, Ctrl+number is the shortcut to add a tuplet, so we can't use it for switching tabs while a note or rest is selected. Since that's true most of the time, I think that means we can't use it at all unfortunately. Windows key+number is also not available as that's used by the OS.

What could work is Shift+Tab to focus the tab bar, then press a number to focus a tab, or even press the first letter of the tab name (e.g. 'V' for 'Violin'). If you keep pressing 'V' it could cycle between the tabs for Violin, Viola, and Violincello.

**Comment by avvvvve** [comment]
Just noting that on Mac, the commands to switch tabs in Chrome (and other browsers) are a little different. [Source](https://support.google.com/chrome/answer/157179?sjid=14303897353572701522-NA)

### Windows
<img width="680" alt="Image" src="https://github.com/user-attachments/assets/bf4f3104-4b24-4707-9f6e-7916c0068240" />

### Mac
<img width="679" alt="Image" src="https://github.com/user-attachments/assets/286041a8-597b-4ac6-98f4-45048c0ecd98" />

The Mac option doesn't pose the same conflicts with tab navigation that the Windows shortcuts do, but it _does_ conflict with nudging certain elements on the score when selected, so unlike the Windows shortcut it should probably only work when nothing in the score is selected.

And Peter, I support your suggestions to:

- make the Tab key visit every control
- add functionality to go to tabs/UI elements that start with the letter key you've pressed

**Comment by shoogle** [comment]
We'e already using Ctrl+number as the shortcut to create tuplets (e.g. Ctrl+3 enters a triplet) so let's not use those shortcuts to switch to specific tabs. We could still have "Switch to tab N" actions in Preferences > Shortcuts, but we can leave it to the user to define their own shortcuts for those actions.

I agree we should implement Ctrl+Tab and Ctrl+Shift+Tab to switch to the next or previous tab (that's Ctrl on all platforms; _not_ Cmd on macOS). This should only happen when the score has focus. Outside the score those shortcuts should continue to function as they do now.
